docs/wiki/tutorials/Pluggable Functions Tutorial.md

```diff
@@ -0,0 +1,128 @@
+
+Here's an example of a `Plugin` implementation:
+
+```Kotlin
+package org.partiql.plugins.mockdb
+
+import org.partiql.spi.Plugin
+import org.partiql.spi.connector.Connector
+import org.partiql.spi.function.PartiQLFunction
+
+public class LocalPlugin implements Plugin {
+override fun getConnectorFactories(): List<Connector.Factory> = listOf()
+
+    @PartiQLFunctionExperimental
+    override fun getFunctions(): List<PartiQLFunction> = listOf(
+        TrimLead // Specify the functions
+    )
+}
+```
+
+## Step 4: Create Service Provider Configuration file
+
+In order for the Java's ServiceLoader to recognize your plugin and load your custom functions, you'll need to specify your `Plugin` implementation in a Service Provider Configuration file:
+
+1. In your project's main resources directory (usually `src/main/resources`), create a new directory named `META-INF/services`.
+
+2. Within this directory, create a new file named after the full interface name as `org.partiql.spi.Plugin`.
+
+3. Inside this file, write the fully qualified name of your Plugin implementation class. If you have multiple implementations, each should be listed on a new line.
+
+Here's an example if your `Plugin` implementation class is `org.partiql.plugins.mockdb.LocalPlugin`:
+```Kotlin
+org.partiql.plugins.mockdb.LocalPlugin
+```
+
+
+
+## Step 5: Package Your Functions into a .jar File
+
+Compile your function and `Plugin` implementation into bytecode and package it into a .jar file. This .jar file will act as a plugin for the PartiQL CLI.
+
+For example, if you are using Gradle, you can simply run `./gradlew build` to compile your module. And your .jar file can be found under `build/libs`.
+
+
+
+## Step 6: Load the Functions into CLI
+
+Each of your .jar files should be stored in its own subdirectory under the `plugins` directory, which itself is inside the .partiql directory in your home directory. Here's what the directory structure should look like:
+
+```
+~/.partiql/plugins
+    ├── firstPlugin
+    │   └── firstPlugin.jar
+    └── secondPlugin
+        └── secondPlugin.jar
+```
+
+In the example above, `firstPlugin.jar` and `secondPlugin.jar` are the plugins you've created, each in its own directory under `~/.partiql/plugins`.
+
+By default, the PartiQL CLI will search the `~/.partiql/plugins` directory for plugins. However, you can specify a different directory when starting the CLI with the `--plugins` option:
+
+```shell
+partiql --plugins /path/to/your/plugin/directory
+```
+
+With this command, the CLI will search for .jar files in the specified directory’s subdirectories and load them as plugins. This feature gives you the flexibility to organize your plugins in a manner that best suits your needs.
```

Comment:
Could be useful to show how to invoke your functions at the end of this document.